### PR TITLE
Abstract data store

### PR DESCRIPTION
**Pull request by ekiwi**
These four commits try to decouple the concrete implementation of the `DataStore` from the rest of `treadle`. The eventual goal of my work is to be able to swap out the data type that is used for simulations.

It would be great if you could have a look these first steps, @chick .

### CODE REVIEW

File: src/main/scala/treadle/executable/DataStore.scala [hunk 1]
```diff
@@ -621,24 +616,62 @@ object DataStore {
...
+  val plugins : mutable.HashMap[String, DataStorePlugin]   // TODO: change to getter
+  val activePlugins : mutable.ArrayBuffer[DataStorePlugin] // TODO: change to getter
+
+}
+
+trait DataBuffer extends DataReader {
+  def getTime: Long
+}
+
+trait AbstractDataStore extends DataReader with DataWriter with DataSerializer with DataPluginHost {
```
Comment (edwardcwang) [inline src/main/scala/treadle/executable/DataStore.scala]: Random comment - might be good to have some ScalaDoc here.

Comment (ekiwi) [reply src/main/scala/treadle/executable/DataStore.scala]: Do you think it makes sense to split the different functionality expected from our `AbstractDataStore` into different traits, or should all/some of these be unified directly into `AbstractDataStore`?

**Comment by ekiwi** [comment]
Now rebased on #111

**Comment by ekiwi** [comment]
no longer relevant

**Pull request closed without merging**